idaes/models_extra/power_generation/unit_models/soc_submodels/channel.py

```diff
@@ -0,0 +1,839 @@
+            description="Decides whether or not to create material "
+            "flux terms above or below the channel.",
+        ),
+    )
+
+    def build(self):
+        super().build()
+
+        # Set up some sets for the space and time indexing
+        dynamic = self.config.dynamic
+        tset = self.flowsheet().config.time
+        self.component_list = pyo.Set(
+            initialize=self.config.component_list,
+            ordered=True,
+            doc="Set of all components present in submodel",
+        )
+        # Set up node and face sets and get integer indices for them
+        izfaces, iznodes = common._face_initializer(
+            self, self.config.control_volume_zfaces, "z"
+        )
+        # Space saving aliases
+        comps = self.component_list
+        include_temp_x_thermo = self.config.include_temperature_x_thermo
+
+        common._create_if_none(self, "length_z", idx_set=None, units=pyo.units.m)
+        common._create_if_none(self, "length_y", idx_set=None, units=pyo.units.m)
+
+        common._create_thermal_boundary_conditions_if_none(self, thin=False)
+
+        if not self.config.below_electrode:
+            self.conc_mol_comp_deviation_x0 = pyo.Var(
+                tset,
+                iznodes,
+                comps,
+                doc="Deviation of concentration at electrode surface from that "
+                "in the bulk channel",
+                initialize=0,
+                units=pyo.units.mol / pyo.units.m**3,
+            )
+            self.material_flux_x0 = pyo.Var(
+                tset,
+                iznodes,
+                comps,
+                doc="Material flux from electrode surface to channel "
+                "(positive is in)",
+                initialize=0,
+                units=pyo.units.mol / pyo.units.m**2 / pyo.units.s,
+            )
+
+            @self.Expression(tset, iznodes, comps)
+            def conc_mol_comp_deviation_x1(b, t, iz, j):
+                return 0
+
+            @self.Expression(tset, iznodes, comps)
+            def material_flux_x1(b, t, iz, j):
+                return 0
+
+        else:
+            self.conc_mol_comp_deviation_x1 = pyo.Var(
+                tset,
+                iznodes,
+                comps,
+                doc="Deviation of concentration at electrode surface from that "
+                "in the bulk channel",
+                initialize=0,
+                units=pyo.units.mol / pyo.units.m**3,
+            )
+            self.material_flux_x1 = pyo.Var(
+                tset,
+                iznodes,
+                comps,
+                doc="Material flux from channel to electrode surface "
+                "(positive is out)",
+                initialize=0,
+                units=pyo.units.mol / pyo.units.m**2 / pyo.units.s,
+            )
+
+            @self.Expression(tset, iznodes, comps)
+            def conc_mol_comp_deviation_x0(b, t, iz, j):
+                return 0
```

Comment:
It has to do with the `below_electrode` flag. In one channel, the electrode is located at the x=1 boundary, while in the other it is located at the x=0 boundary. For the side of the channel not facing the electrode, there is no need to create `conc_mol_comp_deviation` and `material_flux` variables, but other equations might expect them to exist (`material_flux` shows up in the material balance, I don't think `conc_mol_comp_deviation` shows up in another equation). I preferred creating `Expressions` with value zero rather than write conditional logic in other places that expect them to exist.